### DISCUSSION TITLE
Encrypted class names

### DISCUSSION THREAD

**Discussion by ekologic** [vfsfitvnm/frida-il2cpp-bridge#366] [General]
Hey guys, I am just wondering do you also get encrypted names after dumping the game. For example in dnSpy:
<img width="1130" alt="image" src="https://github.com/vfsfitvnm/frida-il2cpp-bridge/assets/13753409/d18510ee-4e77-443e-89c8-ba11bef51a52">
I am just curious what this method is called, since main game logic lives in the `-` namespace, all classes there have this weird namings

And If your moding/reversing with these kinds of classes, any tips or best practices for tracing is welcomed :)

**Comment by vfsfitvnm** [comment]
It's an obfuscation technique (e.g. https://www.beebyte.co.uk/), where the original identifiers are replaced with random strings at build time. There's no way to retrieve the original ones, unless the developers have mistakenly included the mapping file that is usually generated for debugging purposes.

However, you could download an older unobfuscated version of the game (if any), dump it, and make educated guesses...
